src/components/UI/Card/CartCards/CartCards.jsx

```diff
@@ -1,88 +1,137 @@
+                    authorization: localStorage.getItem("feetz")
+                }
+            }).then((res) => {
+                console.log(res);
+                if (res.status === 201) {
+                    setWishListState(res.data.wishlist);
+                }
+            }).catch((error) => {
+                console.log(error.message);
+            });
+        } catch (err) {
+            console.log("error ", err.message);
+        }
+    }
+
+    const UpdateCartHandler = async (stateQuantity) => {
+        console.log(cartState.find((cartitem) => cartitem._id === _id));
+        let cartItem = cartState.find((cartitem) => cartitem._id === _id);
+        if (stateQuantity === 'increment' && (cartItem ?. qty < 4)) {
+            setCartState(await UpdateCartService(stateQuantity, _id));
         }
-        catch(err){
-            console.log("error ",err.message);
+
+        if (stateQuantity === 'decrement' && (cartItem ?. qty >= 2)) {
+
+            setCartState(await UpdateCartService(stateQuantity, _id));
         }
     }
     return (
         <>
             <div className="card cart-card cart-card-horizontal cart-banner-stretch">
-                <img className="card-img" src={productImage} alt={author}  loading="lazy" />
+                <img className="card-img"
+                    src={productImage}
+                    alt={author}
+                    loading="lazy"/>
                 <div className="cart-card-horizontal-layout">
                     <div className="card-content">
                         <div className="card-body">
-                            {title}
-                        </div>
+                            {title} </div>
                         <div className="card-body">
                             <h2>₹{price}</h2>
-                    <span className="text-grey">{discount}</span>
-                        </div>   
-                        <div className="card-body">
-                            Quantity
-                            <select>
-                                <option>1</option>
-                                <option>2</option>
-                                <option>3</option>
-                                <option>4</option>
-                            </select>
+                            <span className="text-grey">
+                                {discount}</span>
                         </div>
-                    </div> 
+                        <div className="card-footer-view">
+                        <button className='btn-addToCart'>
+                            {
+                           <div  className='cart-quantity-container'>
+                                <span onClick={
+                                    () => UpdateCartHandler("increment")
+                                }><IcRoundPlus/></span>
+                                {
+                                {...cartState.find((cartitem) => cartitem._id === _id)}?.qty}
```

Comment:
Handle the case of what happens if optional chaining actually returns undefined.